receiver/splunkhecreceiver/splunkhec_to_metricdata.go

```diff
@@ -0,0 +1,175 @@
+
+		for _, metricName := range metricNames {
+			pointTimestamp := convertTimestamp(event.Time)
+			metric := pdata.NewMetric()
+			metric.InitEmpty()
+			metric.SetDataType(pdata.MetricDataTypeNone)
+			metric.SetName(metricName)
+
+			metricValue := values[metricName]
+			if i, ok := metricValue.(int64); ok {
+				addIntGauge(pointTimestamp, i, metric, populateLabels)
+			} else if i, ok := metricValue.(*int64); ok {
+				addIntGauge(pointTimestamp, *i, metric, populateLabels)
+			} else if f, ok := metricValue.(float64); ok {
+				addDoubleGauge(pointTimestamp, f, metric, populateLabels)
+			} else if f, ok := metricValue.(*float64); ok {
+				addDoubleGauge(pointTimestamp, *f, metric, populateLabels)
+			} else if s, ok := metricValue.(*string); ok {
+				// best effort, cast to string and turn into a number
+				dbl, err := strconv.ParseFloat(*s, 64)
+				if err != nil {
+					numDroppedTimeSeries++
+					logger.Debug("Cannot convert metric",
+						zap.String("metric", metricName))
+				} else {
+					addDoubleGauge(pointTimestamp, dbl, metric, populateLabels)
+				}
+			} else if s, ok := metricValue.(string); ok {
+				// best effort, cast to string and turn into a number
+				dbl, err := strconv.ParseFloat(s, 64)
+				if err != nil {
+					numDroppedTimeSeries++
+					logger.Debug("Cannot convert metric",
+						zap.String("metric", metricName))
+				} else {
+					addDoubleGauge(pointTimestamp, dbl, metric, populateLabels)
+				}
+			} else {
+				// drop this point as we do not know how to extract a value from it
+				numDroppedTimeSeries++
+				logger.Debug("Cannot convert metric",
+					zap.String("metric", metricName))
+			}
+
+			if metric.DataType() != pdata.MetricDataTypeNone {
+				metrics.Metrics().Append(metric)
+			}
+		}
+		if metrics.Metrics().Len() > 0 {
+			resourceMetrics.InstrumentationLibraryMetrics().Append(metrics)
+			md.ResourceMetrics().Append(resourceMetrics)
+		}
+	}
+
+	return md, numDroppedTimeSeries
+}
+
+func addIntGauge(ts pdata.TimestampUnixNano, value int64, metric pdata.Metric, populateLabels func(pdata.StringMap)) {
+	intPt := pdata.NewIntDataPoint()
+	intPt.InitEmpty()
+	intPt.SetTimestamp(ts)
+	intPt.SetValue(value)
+	populateLabels(intPt.LabelsMap())
+	metric.SetDataType(pdata.MetricDataTypeIntGauge)
+	metric.IntGauge().InitEmpty()
+	metric.IntGauge().DataPoints().Append(intPt)
+}
+
+func addDoubleGauge(ts pdata.TimestampUnixNano, value float64, metric pdata.Metric, populateLabels func(pdata.StringMap)) {
+	doublePt := pdata.NewDoubleDataPoint()
+	doublePt.InitEmpty()
+	doublePt.SetTimestamp(ts)
+	doublePt.SetValue(value)
+	populateLabels(doublePt.LabelsMap())
+	metric.SetDataType(pdata.MetricDataTypeDoubleGauge)
+	metric.DoubleGauge().InitEmpty()
+	metric.DoubleGauge().DataPoints().Append(doublePt)
+}
+
+func convertTimestamp(sec float64) pdata.TimestampUnixNano {
+	if sec == 0 {
+		return 0
+	}
+	return pdata.TimestampUnixNano(sec * 1e9)
+}
+
+func buildLabelKeysAndValues(
```

Comment:
Add a comment to explain what this func does, it looks non trivial and unclear what it returns.